pkg/installmanager/installmanager.go

```diff
@@ -853,7 +851,7 @@ func (m *InstallManager) generateAssets(cd *hivev1.ClusterDeployment, mapPoolsBy
 	// For details, see HIVE-1802.
 	// Configure Hive MachineSet manifests with a label for the MachinePool name
 	var awsClient awsclient.Client
-	var vpcID string
+	vpcID := ""
 	addSecurityGroup := (workerMachinePool != nil) && (metav1.HasAnnotation(workerMachinePool.ObjectMeta, constants.ExtraWorkerSecurityGroupAnnotation))
```

Comment:
nit: Seems like we could get rid of the `addSecurityGroup` bool and just use the non-empty-ness of `vpcID` to indicate same. It would be one less arg to pass into the already-cluttered patchMachineSetManifest func.

Similarly, I feel like we could get rid of `addMPLabel`. Inline `len(mapPoolsByType) > 0` on L874 and then just let the `exists` on L1014 filter from there.